src/php/ext/grpc/php_grpc.c

```diff
@@ -206,9 +206,8 @@ void postfork_child() {
   grpc_init();
   grpc_php_init_completion_queue(TSRMLS_C);
 
-  // re-create grpc_channel and point wrapped to it
-  // unlock wrapped grpc channel mutex
-  restart_channels();
+  // clean all channels in the persistent list
+  php_grpc_clean_persistent_list(TSRMLS_C);
```

Comment:
Instead of doing this here, perhaps we should do it slightly earlier in L193: https://github.com/grpc/grpc/blob/master/src/php/ext/grpc/php_grpc.c#L193

Also, we might be missing a `release_persistent_locks()` call in `postfork_child()`. We call `acquire_persistent_locks()` at `prefork_parent()`, and then `release_persistent_locks()` in `postfork_parent()`, but never in `postfork_child()`. Calling `zend_hash_clean()` without `release_persistent_locks()` will result in a hang. So we probably should it around L193 too. So I think the sequence should be

```
... in postfork_child()...

destroy_grpc_channels();

release_persistent_locks();

// OR a function to abstract out 2 callers within php_grpc.c
zend_hash_clean(&grpc_persistent_list);
zend_hash_destroy(&grpc_persistent_list);
zend_hash_clean(&grpc_target_upper_bound_map);
zend_hash_destroy(&grpc_target_upper_bound_map);
```

Reply:
comment addressed